### ISSUE TITLE
Improve AsyncActionCreator type signature to handle optional cancel property correctly

### ISSUE THREAD

**Issue by piotrwitek** [piotrwitek/typesafe-actions#153]
[<img alt="Issuehunt badges" src="https://img.shields.io/badge/IssueHunt-%2440%20Funded-%2300A156.svg" />](https://issuehunt.io/r/piotrwitek/typesafe-actions/issues/153)

## Is your feature request related to a real problem or use-case?
Fix cancel is never #152

## Describe a solution including usage in code example
Improve the resulting type signature of `AsyncActionCreator` making it more consistent with the rest of the API surface and to handle optional cancel property correctly.

The resulting signature must be looking like this:
```ts
AsyncActionCreator<["FETCH_USERS_REQUEST", undefined], ["FETCH_USERS_SUCCESS", User[]], ["FETCH_USERS_FAILURE", Error], ["FETCH_USERS_CANCEL", string]>
```

## Who does this impact? Who is this for?

All

## Describe alternatives you've considered (optional)

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Sponsors (Total: $40.00)

- [<img src='https://avatars3.githubusercontent.com/u/44827199?v=4' alt='issuehunt' width=24 height=24> issuehunt](https://issuehunt.io/u/issuehunt) ($40.00)

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/piotrwitek/typesafe-actions/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

**Comment by piotrwitek** [comment]
blocking because there is a significant amount of changes in the `next` branch (currently heavily WIP) that needs to be completed before possible to fix that. There is also a pretty high chance it will be fixed by #143

**Comment by piotrwitek** [comment]
Resolved by #194
